src/main/java/doit/shop/controller/Product/CategoryController.java

```diff
@@ -0,0 +1,39 @@
+package doit.shop.controller.Product;
+
+import doit.shop.controller.Product.dto.CategoryUpdateRequest;
+import doit.shop.controller.Product.dto.CategoryUpdateResponse;
+import doit.shop.controller.Product.dto.CategoryRegisterRequest;
+import doit.shop.controller.Product.dto.CategoryRegisterResponse;
+import doit.shop.service.CategoryService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api")
+public class CategoryController {
+
+    private final CategoryService categoryService;
+
+    @GetMapping("/products/categories")
+    public List<CategoryRegisterResponse> getCategories() {
+        return categoryService.getAllCategories();
+    }
+
+    @PostMapping("/categories")
+    public CategoryRegisterResponse registerCategory(@RequestBody CategoryRegisterRequest categoryRegisterRequest) {
+        return categoryService.registerCategory(categoryRegisterRequest);
+    }
+
+    @PutMapping("/categories/{categoryId}")
+    public CategoryUpdateResponse modifyCategory(@PathVariable Long categoryId, @RequestBody CategoryUpdateRequest categoryUpdateRequest) {
+        return categoryService.updateCategory(categoryId, categoryUpdateRequest);
+    }
+
+    @DeleteMapping("/{categoryId}")
```

Comment:
```suggestion
    @DeleteMapping("/categories/{categoryId}")
```
여러 상황들을 고려하였을 때, 위처럼 수정하는 게 어떨까요?
만약 변경하지 않은 채로 클라이언트 측에서 api 요청 보내는 엔드포인트를 몇 개 살펴보면 아래와 같습니다.
- `/api/1` -> 카테고리 id가 1인 카테고리 삭제
- `/api/2` -> 카테고리 id가 2인 카테고리 삭제

이렇게만 보면 api 요청이 어떤 것을 의미하는지 모호해질 것 같아서, 앞에 어떤 종류의 데이터를 다루는지가 endpoint에 들어가는 것이 RESTful한 API에 조금 더 가까워질 것 같습니다!

Reply:
의견 감사합니다!!